File: src/core/dirs.rs
use std::path::PathBuf;
use crate::core::error::GitNavigatorError;

pub fn get_config_directory() -> Result<PathBuf, GitNavigatorError> {
    let base = match std::env::consts::OS {
        "linux" | "freebsd" | "netbsd" | "openbsd" => {
            std::env::var("XDG_CONFIG_HOME")
                .map(PathBuf::from)
                .unwrap_or_else(|_| dirs::home_dir().unwrap_or_default().join(".config"))
        },
        "macos" => {
            dirs::home_dir()
                .unwrap_or_default()
                .join("Library/Application Support")
        },
        "windows" => {
            dirs::config_dir().unwrap_or_default()
        },
        _ => dirs::config_dir().unwrap_or_default(),
    };
    
    Ok(base.join("git-navigator"))
}

pub fn get_cache_directory() -> Result<PathBuf, GitNavigatorError> {
    let base = match std::env::consts::OS {
        "linux" | "freebsd" | "netbsd" | "openbsd" => {
            std::env::var("XDG_CACHE_HOME")
                .map(PathBuf::from)
                .unwrap_or_else(|_| dirs::home_dir().unwrap_or_default().join(".cache"))
        },
        "macos" => {
            dirs::home_dir()
                .unwrap_or_default()
                .join("Library/Caches")
        },
        "windows" => {
            dirs::cache_dir().unwrap_or_default()
        },
        _ => dirs::cache_dir().unwrap_or_default(),
    };
    
    Ok(base.join("git-navigator"))
}

File: internal/core/dirs.go
package core

import (
	"os"
	"path/filepath"
	"runtime"
)

func GetConfigDir() (string, error) {
	base := getConfigBase()
	dir := filepath.Join(base, "git-navigator")
	if err := os.MkdirAll(dir, 0755); err != nil {
		return "", err
	}
	return dir, nil
}

func GetBackupDir() (string, error) {
	configDir, err := GetConfigDir()
	if err != nil {
		return "", err
	}
	backupDir := filepath.Join(configDir, "backups")
	if err := os.MkdirAll(backupDir, 0755); err != nil {
		return "", err
	}
	return backupDir, nil
}

func getConfigBase() string {
	goos := runtime.GOOS
	switch goos {
	case "linux", "freebsd", "netbsd", "openbsd", "darwin":
		if xdg := os.Getenv("XDG_CONFIG_HOME"); xdg != "" {
			return xdg
		}
		home, _ := os.UserHomeDir()
		if home != "" {
			if goos == "darwin" {
				return filepath.Join(home, "Library", "Application Support")
			}
			return filepath.Join(home, ".config")
		}
		return "~/.config"
	case "windows":
		if appdata := os.Getenv("APPDATA"); appdata != "" {
			return appdata
		}
		return "."
	default:
		if xdg := os.Getenv("XDG_CONFIG_HOME"); xdg != "" {
			return xdg
		}
		home, _ := os.UserHomeDir()
		if home != "" {
			return filepath.Join(home, ".config")
		}
		return "~/.config"
	}
}
